src/dndbeyond/content-scripts/character.js

```diff
@@ -277,9 +294,84 @@ function rollAbilityOrSavingThrow(paneClass, rollType) {
     if (character.hasClassFeature("Trance of Order") && character.getSetting("sorcerer-trance-of-order", false))
             roll_properties.d20 = "1d20min10";
 
+    // effects 
+    if(rollType == "saving-throw" && character.getSetting("effects-bless", false)) {
+        roll_properties["modifier"] += "+1d4";
+        addEffect(roll_properties, "Bless");
+    } else if(rollType == "saving-throw" && character.getSetting("effects-bane", false)) {
+        roll_properties["modifier"] += "-1d4";
+        addEffect(roll_properties, "Bane");
+    }
+
+    if((rollType == "saving-throw" || rollType == "ability") && roll_properties.ability == "STR" && character.getSetting("effects-enlarge", false)) {
+        adjustRollAndKeyModifiersWithAdvantage(roll_properties);
+        addEffect(roll_properties, "Enlarge");
+    } else if((rollType == "saving-throw" || rollType == "ability") && roll_properties.ability == "STR" && character.getSetting("effects-reduce", false)) {
+        adjustRollAndKeyModifiersWithDisadvantage(roll_properties);
+        addEffect(roll_properties, "Reduce");
+    }
+    
     return sendRollWithCharacter(rollType, "1d20" + modifier, roll_properties);
 }
 
+function adjustRollAndKeyModifiersWithAdvantage(roll_properties) {
+    // Adjust roll_properties["advantage"]
+    const advantageMapping = {
+        [RollType.NORMAL]: RollType.ADVANTAGE,
+        [RollType.DISADVANTAGE]: RollType.NORMAL,
+        [RollType.OVERRIDE_DISADVANTAGE]: RollType.NORMAL,
+        [RollType.SUPER_DISADVANTAGE]: RollType.DISADVANTAGE
+    };
+
+    if (roll_properties["advantage"] === undefined) {
+        roll_properties["advantage"] = RollType.ADVANTAGE;
+    } else if (roll_properties["advantage"] in advantageMapping) {
+        roll_properties["advantage"] = advantageMapping[roll_properties["advantage"]];
+    }
+    
+    if(key_modifiers.normal_roll) {
```

Comment:
Unfortunately, you can't override the key modifiers here, because if the hotkey is `N` for normal and `A` for advantage, and you switch it, then when the user releases the letter `N`, the `key_modifiers.normal_roll` will get set to `false` (which it already was) and `key_modifiers.advantage` will not be modified, since the letter `A` was never pressed by the user... and you end up with a "stuck hotkey".